### PR TITLE
Part 1 of multiple address support

### PR DESCRIPTION
**Pull request by wwahammy**
Closes https://github.com/CommitChange/tix/issues/3277

Depends on #100

### CODE REVIEW

**Comment by wwahammy** [comment]
@clarissalimab I think this works but I'm still a little hesitant since it involves such a major part of the app. Could you review it and see if you see anything?

File: app/models/supporter.rb [hunk 1]
```diff
@@ -55,6 +65,9 @@ class Supporter < ActiveRecord::Base
   belongs_to :merged_into, class_name: 'Supporter', :foreign_key => 'merged_into'
   has_many :merged_from, class_name: 'Supporter', :foreign_key => "merged_into"
 
+  has_many :addresses, class_name: "SupporterAddress"
+  belongs_to :primary_address, class_name: "SupporterAddress"
```
Comment (clarissalimab) [inline app/models/supporter.rb]: Is that relation correct?

File: spec/lib/insert/insert_import_spec.rb [hunk 2]
```diff
@@ -251,7 +326,7 @@
   #     desigs = @donations.map{|d| d['designation']}
   #     expect(desigs.sort).to eq(Hamster::Vector["third party event", "third party event", "third party event", "third party event"])
   #   end
-
+  next
```
Comment (clarissalimab) [inline spec/lib/insert/insert_import_spec.rb]: Was that an accident?

File: spec/models/multiple_addresses_spec.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        country: "United States",
+        zip_code: "54911"
+      }
+    end
+
+    context "where supporter has no address" do 
+      it {
+        
+      }
+    end
```
Comment (clarissalimab) [inline spec/models/multiple_addresses_spec.rb]: What happens? haha

**Review by clarissalimab — COMMENTED** [review]
Looks great! I just left some questions about a few details :speech_balloon::

File: app/models/supporter.rb (hunk 1 above)
Comment (wwahammy) [reply app/models/supporter.rb]: I think so. It's very weird, isn't it?

File: spec/models/multiple_addresses_spec.rb (hunk 3 above)
Comment (wwahammy) [reply spec/models/multiple_addresses_spec.rb]: Yikes, it looks like I need to test that more!

File: spec/lib/insert/insert_import_spec.rb (hunk 2 above)
Comment (wwahammy) [reply spec/lib/insert/insert_import_spec.rb]: It was apparently! I'm going to remove all that commented stuff to clear it up too while we're here.

Comment (wwahammy) [reply spec/lib/insert/insert_import_spec.rb]: I addressed this in: https://github.com/CommitChange/houdini/pull/104

File: app/models/supporter.rb (hunk 1 above)
Comment (clarissalimab) [reply app/models/supporter.rb]: Ya, I thought it should be `has_one` instead.

**Review by clarissalimab — APPROVED** [review]
Great work! :rocket: I didn't see anything that may cause problems.

File: app/models/supporter.rb (hunk 1 above)
Comment (wwahammy) [reply app/models/supporter.rb]: I always have trouble remembering whether it's `belongs_to` or `has_one`. The diagrams at https://guides.rubyonrails.org/v4.2/association_basics.html actually help me remember. The gist is that `belongs_to` is used on a class when they have a field holding association (in this case, Supporter has a `primary_address_id` field). `has_one` would be used when the class at the other end of the association has the field.

**Pull request merged**